### PR TITLE
Remove dask and distributed dependencies

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*
- Remove dask and distributed dependencies
- This is to check if they are actually needed

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/setup.py [hunk 1]
```diff
@@ -55,6 +55,7 @@
     "openmim>0.1.5,<=0.2.1",
     "defusedxml>=0.7.1,<=0.7.1",
     "albumentations>=1.1.0,<=1.2.0",
+    "jinja2>=3.0.3,<3.2",
```
Comment (zhiqiangdon) [inline multimodal/setup.py:58]: Where do we use `jinja2`?

Comment (Innixma) [reply multimodal/setup.py:58]: https://github.com/autogluon/autogluon/blob/master/multimodal/src/autogluon/multimodal/data/templates.py#L18

Comment (Innixma) [reply multimodal/setup.py:58]: You can refer to the CI error prior to me adding it in this PR: https://github.com/autogluon/autogluon/actions/runs/3914783191/jobs/6692268378#step:4:994

Comment (sxjscience) [reply multimodal/setup.py]: This is for supporting prompt source. I’m okay with removing it for now.

**Review by sxjscience — APPROVED** [review]
LGTM

File: multimodal/setup.py (hunk 1 above)
Comment (Innixma) [reply multimodal/setup.py]: @sxjscience if you'd like to remove jinja usage, you'll need to open a separate follow-up PR. I believe jinja was a dependency of dask/distributed, which is why we didn't error in the past, but with dask/distributed removed, now we need to explicitly add it as a dependency to AutoMM

**Comment by tensor-alchemist** [comment]
@Innixma Is dask and ray support completely removed. I am working with more than 10 M unique time series and I have a memory constraints. What is the best way I can achieve distributed learning with AutoGluon?

**Pull request merged**